provisioner/aws-manager.js

```diff
@@ -166,6 +271,254 @@ AwsManager.prototype._filterSpotRequests = function (spotReqs) {
+          plotSpotFulfilment(requestMightHave);
+          keepInTheList = false;
+        }
+      }
+    });
+
+    if (requestAwaiting.iterationCount++ > MAX_ITERATIONS_FOR_STATE_RESOLUTION) {
+      debug('dropping spot request on the floor');
+      keepInTheList = false;
+    }
+
+    return keepInTheList;
+  });
+
+  /*
+    for instances we need to see if there's a transition reason already and if not,
+    we need to store the instance metadata and check on the next iteration whether
+    that instance has an associated state transition reason.  For when there is a
+    spot price termination, we should also look for a spot request in the good and
+    dead ones to see what we bid on the thing.  We should also track what the time
+    was when we found the difference and we should only try for up to, say, 5 times
+    before we give up on trying to figure out why the instance terminated.
+
+    http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_StateReason.html
+  */
+
+  function plotInstanceDeath (instance, time) {
+    // Let's track when the instance is shut down
+    that.reportInstanceTerminated({
+      provisionerId: that.provisionerId,
+      region: instance.Region,
+      az: instance.Placement.AvailabilityZone,
+      instanceType: instance.InstanceType,
+      workerType: instance.WorkerType,
+      id: instance.InstanceId,
+      spotRequestId: instance.SpotInstanceRequestId,
+      time: dateForInflux(time),
+      launchTime: dateForInflux(instance.LaunchTime),
+      stateCode: instance.State.Code,
+      stateMsg: instance.State.Name,
+      stateChangeCode: instance.StateReason.Code,
+      stateChangeMsg: instance.StateReason.Message,
+    });
+
+    if (instance.StateReason.Code === 'Server.SpotInstanceTermination') {
+      debug('We have a spot price floor!');
+      // Let's figure out what we set the price to;
+      var price;
+
+      deadState.requests.forEach(function (request) {
+        if (!price && request.SpotInstanceRequestId === instance.SpotInstanceRequestId) {
+          price = parseFloat(request.SpotPrice, 10);
+        }
+      });
+
+      apiState.requests.forEach(function (request) {
+        if (!price && request.SpotInstanceRequestId === instance.SpotInstanceRequestId) {
+          price = parseFloat(request.SpotPrice, 10);
+        }
+      });
+
+      if (price) {
+        that.reportSpotPriceFloorFound({
+          region: instance.Region,
+          az: instance.Placement.AvailabilityZone,
+          instanceType: instance.InstanceType,
+          time: new Date().toISOString(),
+          price: price,
+          reason: 'instance-spot-killed',
+        });
+      } else {
+        debug('Could not find a price for a spot-price killed instance');
+      }
+    }
+  }
+
+  // Let's handle instance which already have a state reason, or
+  // save them for a future iteration if not
+  differences.instances.forEach(function (instance) {
```

Comment:
again, another great place to start using arrow functions... So you don't have to write `that`.
**example**
```js
differences.instances.forEach(instance => {
  ...
  this...
});
```
Notice that because there is only a single parameter you can write it as `param => {/* body */}`.